lib/Doctrine/DBAL/Migrations/Migration.php

```diff
@@ -95,14 +95,18 @@ public function writeSqlFile($path, $to = null)
 
         $this->outputWriter->write(sprintf("-- Migrating from %s to %s\n", $from, $to));
 
-        $sqlWriter = new SqlFileWriter(
+        return $this->createSqlFileWriter($path)
+                    ->write($sql, $direction);
+    }
+
+    protected function createSqlFileWriter(string $path): SqlFileWriter
```

Comment:
I can't... and it really hurts me.
We were using mockery a big nasty black magic to overload the instantiation of that class. I didn't wan't to do a huge design change to make it work properly with only PHPUnit mocks. 

Suggestion?